src/OptimalTransport.jl

```diff
@@ -184,15 +187,25 @@ Return optimal transport coupling `γ` of the same dimensions as `C` which solve
 
 where ``H`` is the entropic regulariser, ``H(\\gamma) = -\\sum_{i, j} \\gamma_{ij} \\log(\\gamma_{ij})``.
 
+Note that for a common cost `C`, multiple histograms may be provided for a batch computation by passing `mu` and `nu`
+as matrices whose columns `mu[:, i]` and `nu[:, i]` correspond to pairs of histograms. 
+The output in this case is a vector `γ` of coupling matrices such that `γ[i]` is a coupling of `mu[:, i]` and `nu[:, i]`.
+
 """
-function sinkhorn(mu, nu, C, eps; kwargs...)
+function sinkhorn(mu::Vector, nu::Vector, C, eps; kwargs...)
```

Comment:
I don't think it's a good idea to add these type annotations, they are too restrictive. I think it should work if you change `sinkhorn_gibbs` as suggested, then you can change computations depending on the resulting types of `u` and `v`.

Reply:
Ok, I've adopted the suggested changes (see new commits). By 'restrictive' do you mean some of the code is duplicated? I wasn't sure whether checking the type of the output as I commented originally would resolve the problem of type stability (I guess it does then!)

Reply:
No, restrictive referred to the fact that one might want to use different vector types and not only `Vector` (e.g. static `SVector`s, `FillArray`s, etc.). If you dispatch on `Vector` or include checks for `Vector` (same for `Matrix`), all these other types are not covered and will either error or produce incorrect results.

Reply:
Got it!